src/pygaps/characterisation/enth_sorp_whittaker.py

```diff
@@ -97,104 +118,361 @@ def enthalpy_sorption_whittaker(
 
     """
 
-    if isinstance(isotherm, ModelIsotherm):
-        if isotherm.units['pressure_unit'] != 'Pa':
-            raise ParameterError('''Model isotherms should be in Pa.''')
-        model = isotherm.model.name
-    elif isinstance(isotherm, PointIsotherm):
-        isotherm.convert_pressure(unit_to='Pa')
+    # Critical, triple and saturation pressure
+    p_c = isotherm.adsorbate.p_critical()
+    p_t = isotherm.adsorbate.p_triple()
+    p_sat = isotherm.adsorbate.saturation_pressure(
+        temp=isotherm.temperature,
+        pseudo=True,
+    )
+
+    if not isinstance(isotherm, (PointIsotherm, ModelIsotherm)):
+        raise ParameterError(
+            f'''
+            Isotherm must be ModelIsotherm or PointIsotherm.
+            You have input a {type(isotherm)}.
+            '''
+        )
+
+    if (isinstance(isotherm, ModelIsotherm) and isotherm.units['pressure_unit'] != 'Pa'):
+        raise ParameterError('''Model isotherms should be in Pa.''')
+
+    if isinstance(isotherm, PointIsotherm):
+        if model == 'guess':
+            model = pgm._WHITTAKER_MODELS
+
+        isotherm.convert_pressure(unit_to='Pa', mode_to='absolute', pseudo=True)
```

Comment:
By converting the pressure here, the isotherm itself is changed. It's usually not recommended for functions to modify classes that are being passed into it, unless this is specifically the goal of that code.

Reply:
resolved [9b217b7](https://github.com/pauliacomi/pyGAPS/pull/54/commits/9b217b70ed3778b4ee3187801a3cf71d953756b3)